web/v1/snowflake/sql-runner/sql/standard/00-setup/01-main/01-stored-procedures.sql

```diff
@@ -86,74 +86,99 @@ CREATE OR REPLACE PROCEDURE {{.output_schema}}.column_check(SOURCE_SCHEMA VARCHA
+    SELECT SUM(CASE WHEN s.column_name IS NULL THEN 1 ELSE 0 END) AS missing_in_source,
+          SUM(CASE WHEN t.column_name IS NULL THEN 1 ELSE 0 END) AS missing_in_target
+    FROM target_data  t
+    FULL OUTER JOIN SOURCE_data s
+    ON t.ordinal_position = s.ordinal_position AND T.data_type = s.data_type AND s.column_name = t.column_name
+    WHERE t.ordinal_position>= s.ordinal_position OR s.ordinal_position is null),
+
+  columns_to_add AS (
+    SELECT listagg(CASE WHEN t.ordinal_position IS NULL THEN s.column_definition end
+    , ', ') WITHIN GROUP (ORDER BY s.ordinal_position) AS cols_to_add
+    FROM source_data s
+    LEFT JOIN target_data t
+    ON t.ordinal_position = s.ordinal_position )
+
+  SELECT missing_in_source, missing_in_target, cols_with_varchar_issue, cols_to_add
+  FROM columns_to_add, type_check, varchar_check`;
+
+  var res = snowflake.createStatement({sqlText: column_check_stmt,
+                                       binds: [SOURCE_SCHEMA, SOURCE_TABLE,TARGET_SCHEMA, TARGET_TABLE]}
+                                      ).execute();
+  res.next();
+
+  missing_in_source = res.getColumnValue(1);
+  missing_in_target = res.getColumnValue(2);
+  cols_with_varchar_issue = res.getColumnValue(3);
+  cols_to_add = res.getColumnValue(4);
+
+  if (missing_in_target > 1) {
+    throw "ERROR: Source table is missing column(s) which exist in target table.";
   }
 
-  // == Helpers ==
-
-  function list_cols_with_type(sch,tbl,delimiter) {
-      var stmt = `
-          SELECT
-              LISTAGG(
-                  CASE
-                    WHEN isc.data_type='TEXT'
-                      THEN CONCAT(isc.column_name, ' VARCHAR(',isc.character_maximum_length, ')')
-                    WHEN isc.data_type='NUMBER'
-                      THEN CONCAT(isc.column_name, ' NUMBER(', isc.numeric_precision, ',',isc.numeric_scale, ')')
-                    ELSE
-                      CONCAT(isc.column_name, ' ', isc.data_type)
-                  END, '` + delimiter + `')
-              WITHIN GROUP (order by isc.ordinal_position)
-          FROM information_schema.columns AS isc
-          WHERE table_schema='` + sch + `'
-            AND table_name='` + tbl + `';`;
-
-      var res = snowflake.createStatement({sqlText: stmt}).execute();
-      res.next();
-      result = res.getColumnValue(1);
-
-      return result;
+  if (cols_with_varchar_issue !== '') {
+    throw "ERROR: field length for source varchar column(s) " + cols_with_varchar_issue + " is longer than the target."
   }
 
-  function notIncludedIn(arr) {
-      return function(elt) {
-        return ! arr.includes(elt);
-      };
+  if (missing_in_source > 0) {
+    throw "ERROR: Can only migrate extra columns of the end of source"
   }
 
-  function add_columns_to(sch, tbl, cols) {
-      var alter_stmt = `ALTER TABLE ` + sch + `.` + tbl + ` ADD COLUMN ` + cols;
-      snowflake.createStatement({sqlText: alter_stmt}).execute();
+  if (cols_to_add !== '') {
```

Comment:
I think it might be a bit clearer to use `missing_in_target > 0` here. WDYT?

Reply:
Good suggestion, this has been taken into consideration for the final commit.